DedicatedServer/Scripts/Modes/Trackmania/FlagRush.Script.txt

```diff
@@ -799,12 +799,13 @@ Void SpawnPlayers() {
 		if (Player.SpawnStatus == CSmPlayer::ESpawnStatus::NotSpawned) {
 			declare CMapSpawn Spawn = FlagRush_Map::GetSpawn(Player).PlayerSpawn;
 			if (Spawn != Null) {
-				declare Integer SpawnDate for Player;
+				declare CUIConfig PlayerUI = UIManager.GetUI(Player);
+				declare netwrite Integer Net_SpawnDate for PlayerUI;
```

Comment:
I'm not sure if too relevant, but:
SpawnDate is gameplay relevant, not just UI relevant. Therefore I would do the netwrite declare for Player and not PlayerUI.
```
declare netwrite Integer Net_SpawnDate for Player;
```

This way we would also not have to explicitly get the UI as it's done in line 802 right now.
Additionally, I'm not sure if fakeplayers (bots) have UIs, so if they for some reason get to have a Net_SpawnDate at some point, this might lead to a NPE?
Also: By writing it to the player, other clients will also be aware of the respawn timers of other players. This could be usefull if we would want to for example add a UI later where you can see which player is currently spawned and how long other player will have left on there respawn timer.